### PR TITLE
[Task CS-59] Error messages in authorization

### PR DESCRIPTION
**Pull request by twistzzmc**

### CODE REVIEW

**Review by jmysliv — CHANGES_REQUESTED** [review]
Error messages should be in Polish.

File: rest/routes/user.py [hunk 1]
```diff
@@ -43,20 +42,22 @@ def register():
...
     data = {}
-    if len(name) < 3 or len(name) > 30 or re.match('^[.a-zA-Z0-9_-]+$', name) is None:
+    if len(name) < 3 or len(name) > 30 or \
+            re.match(r'^[ąĄćĆęĘłŁńŃóÓśŚźŹżŻ.a-zA-Z0-9 _-]+$', name) is None:
         data['name'] = name
-    if not email.endswith('agh.edu.pl') or re.match('^[.@a-zA-Z0-9_-]+$', email) is None:
+    if not email.endswith('agh.edu.pl') or re.match(r'^[.@a-zA-Z0-9_-]+$', email) is None:
         data['email'] = email
-    if len(password) < 3 or len(password) > 30 or re.match('^[.a-zA-Z0-9_-]+$', password) is None:
+    if len(password) < 3 or len(password) > 30 or re.match(r'^[.a-zA-Z0-9_-]+$', password) is None:
```
Comment (skusnierz) [inline rest/routes/user.py:50]: This regex doesn't include special characters like `!` ...etc. And I think in this app it is not necessary to validate input data with the use of regular expressions.

File: rest/routes/user.py [hunk 2]
```diff
@@ -43,20 +42,22 @@ def register():
...
         data['password'] = password
 
     if len(data) > 0:
+        data['error'] = 'Invalid registration data'
```
Comment (skusnierz) [inline rest/routes/user.py:54]: You should return a more appropriate error message, I mean when the email format is invalid, the backend ought to return `"Wprowadzony email jest niepoprawny"` or something else, the same applies to name and password.

**Review by skusnierz — CHANGES_REQUESTED** [review]

**Review by jmysliv — CHANGES_REQUESTED** [review]
Zmień errory na Polski.

**Review by jmysliv — COMMENTED** [review]
@twistzzmc Looks okay, but in this repository we require linear history, so you CAN'T merge master branch into yours. Revert merge commit and do rebase.

**Review by jmysliv — APPROVED** [review]

**Review by skusnierz — APPROVED** [review]

**Review by twrdyyy — APPROVED** [review]
lgtm

**Pull request merged**